### PR TITLE
Add "split" xpath in post-processing , newlines in replace support

### PR DESCRIPTION
**Pull request by bnkai**
Adds support for split that is the opposite of concat.
It splits a string according to the separator it is given, while the separator is removed.
If some of the resulting strings are empty  eg splitting `tag1,,tag2,tag3` with `,` then they are ignored
Result should be  0:`tag1 ` 1:`tag2` 2:`tag3`

It is executed last in order of processing: concat -> replace -> parseDate -> split

The community scraper for [AdultTimeStudios](https://github.com/stashapp/CommunityScrapers/pull/54) for example can be written like this 
```yaml
name: "AdultTimeStudios"
sceneByURL:
  - action: scrapeXPath
    url:
      - puretaboo.com/
      - isthisreal.com/en/video/
      - 21sextury.com/en/video/

    scraper: sceneScraper
xPathScrapers:
  sceneScraper:
    common:
      $videoscript: //script[contains(text(), 'ScenePlayerId = "player"')]/text()
      $datascript: //script[contains(text(), 'sceneDetails')]/text()
      $imagescript: //script[contains(text(), 'picPreview')]/text()
    scene:
      Title:
        selector: $videoscript
        replace:
          - regex: .+(?:"sceneTitle":")([^"]+).+
            with: $1
          - regex: .+(?:"sceneTitle":"").+
            with:
      Date:
        selector: $videoscript
        replace:
          - regex: .+(?:"sceneReleaseDate":")([^"]+).+
            with: $1
        parseDate: 2006-01-02
      Details:
        selector: $datascript
        replace:
          - regex: .+(?:sceneDescription":")(.+)(?:","sceneActors).+
            with: $1
          - regex: .+(?:"sceneDescription":"").+
            with:
          - regex: <\\\/br>|<br\s\\\/>|<br>
            with: "\n"
      Tags:
        # Section still being worked on
        Name:
          selector: $datascript
          replace:
            - regex: .+(?:sceneCategories":\[)(.+)(?:\],"sceneViews").+
              with: $1
            - regex: \"
              with:
          split: ","
      Performers:
        # Section still being worked on
        Name:
          selector: $datascript
          replace:
            - regex: .+(?:"sceneActors":)(.+)(?:,"sceneCategories").+
              with: $1
            - regex: \d+|actorId|actorName|\[|\]|"|\{|:|\}
              with:
          split: ","
      Image:
        selector: $imagescript
        replace:
          - regex: .+(?:picPreview":")([\w:]+)(?:[\\\/]+)([\w-\.]+)(?:[\\\/]+)(\w+)(?:[\\\/]+)(\d+)(?:[\\\/]+)([\d_]+)(?:[\\\/]+)(\w+)(?:[\\\/]+)(\d+)(?:[\\\/]+)(\d+)(?:[\\\/]+)([\w]+)(?:[\\\/]+)([\w.]+).+
            with: $1//$2/$3/$4/$5/$6/$7/$8/$9/$10
            # if using the transport subdomain, parameters need to be passed
            # otherwise a cropped square image is returned by default
          - regex: (https:\/\/transform.+)
            with: $1?width=960&height=543&enlarge=true
```

**EDIT** scraper is still not complete not everything works as it should yet. Its mainly to demonstrate the split and "\n" usage.

### CODE REVIEW

**Comment by bnkai** [comment]
Added support for newlines when they are added with Replace with:
If you add a regex with a `with: "\n"` clause then those newlines should be preserved.
It works by replacing "\n" with "\r" before commonPostprocessing is applied and restores them later.
Due to the commonPostprocessing applied in several places this was the simplest way i could implement this.
The only edge case is if there already "\r" in some fields but i didn't encounter that. It could be altered to use some other special char instead of "\r" if needed.

Tested with the scraper above and the isthisreal.com site mainly and also with  already existing scrapers to make sure no regression occurs.

File: pkg/scraper/xpath.go [hunk 1]
```diff
@@ -299,15 +348,26 @@ func (s xpathScraperConfig) process(doc *html.Node, common commonXPathConfig) xP
...
-					const i = 0
-					ret = ret.setKey(i, k, result)
+					if attrConfig.hasSplit() {
+						for j, txt := range attrConfig.splitString(result) {
+							ret = ret.setKey(j, k, txt)
+						}
+					} else {
+						const i = 0
+						ret = ret.setKey(i, k, result)
+					}
```
[the remark this answers is not in the document]
Comment (bnkai) [reply pkg/scraper/xpath.go]: forgot to refactor
done

**Comment by bnkai** [comment]
```yaml
name: "AngelaWhite"
sceneByURL:
  - action: scrapeXPath
    url:
      - angelawhite.com/tour/trailers/
    scraper: sceneScraper
xPathScrapers:
  sceneScraper:
    common:
      $info: //div[@class="video-details-inner"]
      $search: //div[@class="video-details-inner"]/h2/text()
    scene:
      Title: $info/h2/text()
      Date:
        selector: $info/span/text()
        parseDate: Jan 2, 2006
      Details: $info//p/text()
      Performers:
        Name:
          selector: $search
          replace:
            - regex: (^\d+)\s.+
              with: "http://angelawhite.com/tour/search?query=$1"
          subScraper:
            selector: //div[@class="models-list"]/a/text()
            concat: ","
          split: ","
      Tags:
        Name:
          selector: $search
          replace:
            - regex: (^\d+)\s.+
              with: "http://angelawhite.com/tour/search?query=$1"
          subScraper:
            selector: //div[@class="categories-list"]/a/text()
            concat: ","
          split: ","
      Image:
        selector: $search
        replace: 
          - regex: \s
            with: "+"
          - regex: ^
            with: "http://angelawhite.com/tour/search?query=\""
          - regex: $
            with: "\""
        subScraper:
          selector: //img/@src0
          replace:
            - regex: ^
              with: http://angelawhite.com
# Last Updated June 17, 2020
```
The above is another sample scraper that demonstrates/uses the `split` function.

**Comment by WithoutPants** [comment]
Don't forget to update the wiki once this is merged.

**Pull request merged**